Created a skeleton of the documentation (#27)
populationgenomics/structural-constraint: docs/alphafold_transcript_coverage.md

# Abstract

> The AlphaFold/AlphaFold2 database from EMBL-EBI is comprised of a single protein structure per gene symbol. These are named based on UniProt accessions with identical sequence to each AlphaFold structure, allowing precise mapping of sequence-based information to the protein tertiary structures. While useful, there is enormous clinical value in building structures from alternative transcripts for each gene symbol, where we have knowledge that these are protein-coding and expressed. Often, alternate transcripts are of significant relevance in specific tissues, such as the brain or immune cells and of specific interest to scRNA-Seq experiments. We explore the coverage of the protein-coding space in the human genome compared to the AlphaFold structures and show that 11.8% of actionable ClinVar variants are currently not represented and contained within alternate transcripts.  

Reviewer: ```suggestion
The AlphaFold/AlphaFold2 database from EMBL-EBI is comprised of a single protein structure per gene symbol. These are named based on UniProt accession numbers, with identical sequences to each AlphaFold structure; this allows for precise mapping of sequence-based information to the protein tertiary structures. While useful, there is enormous clinical value in building structures from alternative transcripts for each gene symbol, where we have knowledge that these are protein-coding and expressed. Often, alternate transcripts are of significant relevance in specific tissues, such as the brain or immune cells, and of specific interest to scRNA-Seq experiments. We explore the coverage of the protein-coding space in the human genome compared to the AlphaFold structures and show that 11.8% of actionable ClinVar variants are currently not represented and contained within alternate transcripts.  
```

Reviewer 2: This reads much better, thanks! (and I love semicolons)

# Steps

> ## Files downloaded

Reviewer: Personally, here I'd lay these steps out numerically, e.g.,
1. Download files
2. Unpack the AlphaFold2 transcripts

etc.

Reviewer 2: Thanks, it's a good idea, I'll edit this as a separate PR after merging the pages branch to main

-   Accessed 9/11/2021

GENCODE protein-coding transcripts  

-   gencode.v39.pc\_transcripts.fa
-   <http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/latest_release/gencode.v39.pc_transcripts.fa.gz>
-   Accessed 20/1/22

GENCODE protein-coding translations  

-   gencode.v39.pc\_translations.fa
-   <http://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/latest_release/gencode.v39.pc_translations.fa.gz>
-   Accessed 20/1/22

GENCODE comprehensive gene annotation (gtf format)  

-   gencode.v39.annotation.gtf
-   <https://ftp.ebi.ac.uk/pub/databases/gencode/Gencode_human/release_39/gencode.v39.annotation.gtf.gz>
-   Accessed 20/1/22

ClinVar variant summary table  

-   variant\_summary.txt.gz
-   <https://ftp.ncbi.nlm.nih.gov/pub/clinvar/tab_delimited/variant_summary.txt.gz>
-   Accessed 25/1/22

## Unpack the AlphaFold2 transcripts

Unzip the AlphaFold2 structures into a folder. In my case, `~/data/af2/`  

    tar -xvf UP000005640_9606_HUMAN.tar

## Get the UniProt ID&rsquo;s corresponding to AF2 structures

Quick shell command to parse the names from the AF2 structure filenames.  

    ls | awk -F"-" '{print $2}' | sort | awk '!a[$0]++' > uniprots.txt

> 20,504 unique UniProt IDs to use.  

Reviewer: Is this just a general comment of how many you found in this example, or is will this stay permanent? If the latter, I'd expand and explain that a bit more formally.

Reviewer 2: Thanks! I'll expand on this in a separate PR after moving pages deployment to main branch

> ## Get UniProt metadata and matched ENSGs

Reviewer: ```suggestion
## Get UniProt metadata and matched Ensembl gene IDs
```

Reviewer: It would be nice to introduce what this is before giving the abbreviation.

> To match the UniProt accessions to GENCODE, I&rsquo;m using the ENSG&rsquo;s paired with each UniProt structure in the UniProtKB idmapping table. For whatever reasons, matching GENCODE ENST&rsquo;s to UniProtKB&rsquo;s listed ENST&rsquo;s directly is very lossy (even disregarding transcript version numbers). Here we&rsquo;re just getting the idmapping table, filtering to UniProts where we have an AlphaFold2 structure and splitting out the ENSG from `ENSG000001234; ENSG000001235` into separate rows.  

Reviewer: ```suggestion
To match the UniProt accessions to GENCODE, I'm using the ENSGs paired with each UniProt structure in the UniProtKB idmapping table. For whatever reasons, matching GENCODE Ensembl transcripts (ENSTs) to UniProtKB's listed ENSTs directly is very lossy (even disregarding transcript version numbers). Here we're just getting the idmapping table, filtering to UniProts where we have an AlphaFold2 structure and splitting out the ENSG from `ENSG000001234; ENSG000001235` into separate rows.  
```

Reviewer 2: I think there's a number of these "&rsquo;" 's in the document resulting from exporting my notebook style doc into .md, I'll see if I can switch off this behaviour

    library(magrittr)
    library(purrr)
    library(seqinr)
    
    ## gencode_transcripts = read.fasta("~/data/gencode/gencode.v39.pc_transcripts.fa", as.string = T)
    gencode_translations = read.fasta("~/data/gencode/gencode.v39.pc_translations.fa", as.string = T)
    
    
    COLNAMES_TRANSLATIONS = c("ensp", "enst", "ensg",
                       "havana_g", "havana_t", "transcript",
                       "gene", "aalength", "sequence")
    
    
    convert_entry = function(a) {
      header = attr(a, "name") %>% strsplit("|", fixed = T) %>% .[[1]]
      s = a %>% as.character %>% toupper
      vals = c(header, s)
      vals %<>% as.list %>% as.data.table
      names(vals) = COLNAMES_TRANSLATIONS
      return(vals)
    }
    
    
    gencode_translations_dt = lapply(gencode_translations, convert_entry) %>%
      rbindlist(use.names = T, fill = T)
    
    fwrite(gencode_translations_dt, "gencode_translations.txt", sep = "\t")

## Match GENCODE transcripts to UniProt&rsquo;s ENSGs

Next confirm that the GENCODE protein-coding transcripts can be matched to the UniProt-associated ENSGs to see how many GENCODE protein-coding sequences have a valid ENSG.  

> ENSGs with a structure: 21,649  

Reviewer: ```suggestion

ENSGs with a structure: 21,649  
```

## Match GENCODE transcripts to UniProt&rsquo;s ENSGs

Next confirm that the GENCODE protein-coding transcripts can be matched to the UniProt-associated ENSGs to see how many GENCODE protein-coding sequences have a valid ENSG.  

> 

Reviewer: ```suggestion
hgjasdg
```
